app/nexus.go

```diff
@@ -0,0 +1,545 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"github.com/temporalio/tcld/protogen/api/cloud/cloudservice/v1"
+	"github.com/temporalio/tcld/protogen/api/cloud/nexus/v1"
+	"github.com/temporalio/tcld/protogen/api/cloud/operation/v1"
+	"github.com/urfave/cli/v2"
+)
+
+type (
+	NexusClient struct {
+		client cloudservice.CloudServiceClient
+		ctx    context.Context
+	}
+	GetNexusClientFn func(ctx *cli.Context) (*NexusClient, error)
+)
+
+func GetNexusClient(ctx *cli.Context) (*NexusClient, error) {
+	ct, conn, err := GetServerConnection(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &NexusClient{
+		client: cloudservice.NewCloudServiceClient(conn),
+		ctx:    ct,
+	}, nil
+}
+
+func (c *NexusClient) getEndpointByName(endpointName string) (*nexus.Endpoint, error) {
+	endpoints, err := c.listEndpoints(endpointName)
+	if err != nil {
+		return nil, err
+	}
+	if len(endpoints) > 0 {
+		return endpoints[0], nil
+	}
+	return nil, fmt.Errorf("endpoint not found") // TODO: if id is used, get might throw NotFound differently. make consistent.
+}
+
+func (c *NexusClient) listEndpoints(endpointName string) ([]*nexus.Endpoint, error) {
+	listResponse := &cloudservice.GetNexusEndpointsResponse{
+		Endpoints: make([]*nexus.Endpoint, 0),
```

Comment:
why not just do 

`var endpoints []*nexus.Endpoint`

and return endpoints, nil ?

Reply:
This was to show an empty array if no endpoints were found.
```
$ tcld nexus endpoint list
[]
```

Otherwise, the response is nil.

Reply:
Changed it to `endpoints := make([]*nexus.Endpoint, 0)`

If we don't initialize it to an empty list, the `tcld nexus endpoint list` command returns `null` instead of `[]`.